### DISCUSSION TITLE
Using a keyPath with unique: true

### DISCUSSION THREAD

**Discussion by nielsnl68** [ujjwalguptaofficial/JsStore#237] [Q&A]
In one of my table i use the following the column: `unique: { keyPath: ['userID', 'date', 'time'], unique: true }` but not sure of this should work, does it?

At the end, at this moment it dont work; you can insert double records with same  userID, date and time.

**Comment by ujjwalguptaofficial** [comment]
what is the data type of userId, date & time ?

**Comment by nielsnl68** [comment]
this is the table:

```javascript
 var metingen = {
        name: 'metingen',
        columns: {
            id: { primaryKey: true, autoIncrement: true },
            userID: { notNull: true, dataType: 'number' },
            date: { notNull: true, dataType: 'string' },
            time: { notNull: true, dataType: 'string' },
            unique: { keyPath: ['userID', 'datum', 'tijd'], unique: true }
        },
        alter: {

        }
    };
```

**Reply by nielsnl68** [reply]
userID is a reference to an other table.

**Reply by ujjwalguptaofficial** [reply]
ok, will check

**Comment by ujjwalguptaofficial** [comment]
@nielsnl68 it works. Your keypath is referencing wrong column - it should be like this :

```
  unique: {
                    keyPath: ['userID', 'date', 'time'],
                    unique: true
                }
```

I have also added test case for this, check this commit - https://github.com/ujjwalguptaofficial/JsStore/commit/d83cf929fbc61463e332695dcd563241045c6cc2

**Reply by nielsnl68** [reply]
Oww ... that was a bad translation of me.. yes it should have been shown as you said.  My mistake, i did not wanted to use the dutch naming but i forgot to change those in the unique.

for me it did not work. i got 2 records with the same values in my db.

**Reply by ujjwalguptaofficial** [reply]
how have you created this unique column ? i mean using alter or direct ? If using alter was it `add` or `modify` ?
